verification/spec/relayer/Relayer.tla

```diff
@@ -0,0 +1,769 @@
+ Connection datagrams
+ ***************************************************************************)    
+\* Compute connection datagrams designated for dstChainID. 
+\* These are used to update the connection end on dstChainID in the environment.
+ComputeConnectionDatagrams(srcChainID, dstChainID) ==
+    LET srcConnectionEnd == GetConnectionEnd(srcChainID) IN
+    LET dstConnectionEnd == GetConnectionEnd(dstChainID) IN
+
+    LET srcConnectionID == GetConnectionID(srcChainID) IN
+    LET dstConnectionID == GetConnectionID(dstChainID) IN
+
+    LET srcHeight == GetLatestHeight(srcChainID) IN
+    LET srcConsensusHeight == GetCounterpartyClientHeight(srcChainID) IN
+    
+    IF dstConnectionEnd.state = "UNINIT" /\ srcConnectionEnd.state = "UNINIT" THEN 
+         {[type |-> "ConnOpenInit", 
+           connectionID |-> dstConnectionID, \* "connBtoA" (if srcChainID = "chainA", dstChainID = "chainB")
+           clientID |-> GetCounterpartyClientID(dstChainID), \* "clA"
+           counterpartyConnectionID |-> srcConnectionID, \* "connAtoB"
+           counterpartyClientID |-> GetCounterpartyClientID(srcChainID)]} \* "clB" 
+    
+    ELSE IF srcConnectionEnd.state = "INIT" /\ \/ dstConnectionEnd.state = "UNINIT"
+                                               \/ dstConnectionEnd.state = "INIT" THEN 
+         {[type |-> "ConnOpenTry",
+           desiredConnectionID |-> dstConnectionID, \* "connBtoA" (if srcChainID = "chainA", dstChainID = "chainB")  
+           clientID |-> srcConnectionEnd.counterpartyClientID, \* "clA"
+           counterpartyConnectionID |-> srcConnectionID, \* "connAtoB"
+           counterpartyClientID |-> srcConnectionEnd.clientID, \* "clB" 
+           proofHeight |-> srcHeight,
+           consensusHeight |-> srcConsensusHeight]}
+         
+    ELSE IF srcConnectionEnd.state = "TRYOPEN" /\ \/ dstConnectionEnd.state = "INIT"
+                                                  \/ dstConnectionEnd.state = "TRYOPEN" THEN
+         {[type |-> "ConnOpenAck",
+           connectionID |-> dstConnectionID, \* "connBtoA" (if srcChainID = "chainA", dstChainID = "chainB")
+           proofHeight |-> srcHeight,
+           consensusHeight |-> srcConsensusHeight]} 
+         
+    ELSE IF srcConnectionEnd.state = "OPEN" /\ dstConnectionEnd.state = "TRYOPEN" THEN
+         {[type |-> "ConnOpenConfirm",
+           connectionID |-> dstConnectionEnd.connectionID, \* "connBtoA" (if srcChainID = "chainA", dstChainID = "chainB")
+           proofHeight |-> srcHeight]} 
+    ELSE {}
+
+\* Get the connection datagrams designated for srcChainID and dstChainID         
+ConnectionDatagrams(srcChainID, dstChainID) ==
+    \* connection datagrams from dstChainID to srcChainID 
+    LET srcConnectionDatagrams == ComputeConnectionDatagrams(dstChainID, srcChainID) IN
+    \* connection datagrams from srcChainID to dstChainID
+    LET dstConnectionDatagrams == ComputeConnectionDatagrams(srcChainID, dstChainID) IN
```

Comment:
If I am not wrong `ConnOpenInit` will be created for both source and destination chain in case `dstConnectionEnd.state = "UNINIT" /\ srcConnectionEnd.state = "UNINIT"`?

Reply:
This is fixed now, we are only sending in one direction, i.e., from `srcChainID` to `dstChainID`